workspaces/website/src/api/index.ts

```diff
@@ -48,3 +53,70 @@ apiRouter.get(
     );
   }
 );
+
+/**
+ * Newsletter subscribe api route
+ */
+
+apiRouter.post(
+  '/newsletter-subscribe',
+  async (req, env: PAGES_VARS) => {
+    try {
+      const formData = new FormData();
+      formData.append('secret', env.CLOUDFLARE_RECAPTCHA_KEY);
+      formData.append('response', req.query.token as string);
+      
+      const captchaUrl = 'https://challenges.cloudflare.com/turnstile/v0/siteverify';
+      const captchaResponse = await fetch(captchaUrl, {
+        body: formData,
+        method: 'POST',
+      });
+
+      const captchaResult = await captchaResponse.json() as { success: boolean };
+
+      if (!captchaResult.success) {
+        return corsify(error(
+          422,
+          { title: 'Invalid Captcha' }
+        ));
+      }
+
+      const mailchimpUrl = `https://${mailchimpServer}.api.mailchimp.com/3.0/lists/${mailchimpList}/members`;
```

Comment:
remove the `workspaces/website/src/api/constants.ts` file and create an env variable like `VITE_MAILCHIMP_NEWSLETTER_URL` where
```tsx
VITE_MAILCHIMP_NEWSLETTER_URL=https://us12.api.mailchimp.com/3.0/lists/a7c9440979/members
```